### PR TITLE
[12.0][MIG] hr_employee_id

### PR DESCRIPTION
**Pull request by alexey-pelykh**

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
@pedrobaeza requesting input: `identification_id` is declared over here [addons/hr/models/hr.py#L141](https://github.com/odoo/odoo/blob/12.0/addons/hr/models/hr.py#L143) right next to SSN & SIN, used over here [addons/hr_payroll/views/report_payslipdetails_templates.xml#L29](https://github.com/odoo/odoo/blob/12.0/addons/hr_payroll/views/report_payslipdetails_templates.xml#L29) as arbitrary Employee ID and used over here [addons/l10n_fr_hr_payroll/report/report_l10n_fr_fiche_paye.xml#L51](https://github.com/odoo/odoo/blob/12.0/addons/l10n_fr_hr_payroll/report/report_l10n_fr_fiche_paye.xml#L51) as SSN (I guess?). And this module redeclares that field as generated sequence. While original usage suggests that it's something like SSN or ITIN or similar. Whom to discuss this with?

File: hr_employee_id/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# © 2011, 2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
```
Comment (nikul-serpentcs) [inline hr_employee_id/__init__.py]: @alexey-pelykh Remove Copyright form all `__init__` file

File: hr_employee_id/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        'Brainbean Apps, '
+        'Michael Telahun Makonnen, '
+        'OpenSynergy Indonesia, '
+        'Camptocamp, '
+        'Odoo Community Association (OCA)',
+    'website': 'https://github.com/OCA/hr',
+    'depends': [
+        'hr'
+    ],
+    'data': [
```
Comment (nikul-serpentcs) [inline hr_employee_id/__manifest__.py:20]: `data` sequence as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L29)

File: hr_employee_id/models/hr_employee.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# © 2011, 2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
+# © 2016 OpenSynergy Indonesia
+# © 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import random
+import string
+from odoo import api, fields, models, _
+from odoo.exceptions import Warning as UserWarning
```
Comment (nikul-serpentcs) [inline hr_employee_id/models/hr_employee.py]: use `UserError`

File: hr_employee_id/views/hr_employee_views.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <record id="view_employee_form" model="ir.ui.view">
+        <field name="name">hr.employee.form.inherit</field>
+        <field name="model">hr.employee</field>
+        <field name="inherit_id" ref="hr.view_employee_form"/>
+        <field name="arch" type="xml">
+            <data>
```
Comment (nikul-serpentcs) [inline hr_employee_id/views/hr_employee_views.xml]: remove it

File: hr_employee_id/views/hr_employee_views.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                </h1>
+            </data>
+        </field>
+    </record>
+    <record id="view_employee_tree" model="ir.ui.view">
+        <field name="name">hr.employee.tree.inherit</field>
+        <field name="model">hr.employee</field>
+        <field name="inherit_id" ref="hr.view_employee_tree"/>
+        <field name="arch" type="xml">
+            <data>
```
Comment (nikul-serpentcs) [inline hr_employee_id/views/hr_employee_views.xml]: same here

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: hr_employee_id/views/hr_employee_views.xml (hunk 5 above)
Comment (alexey-pelykh) [reply hr_employee_id/views/hr_employee_views.xml]: done

File: hr_employee_id/views/hr_employee_views.xml (hunk 4 above)
Comment (alexey-pelykh) [reply hr_employee_id/views/hr_employee_views.xml]: done

File: hr_employee_id/models/hr_employee.py (hunk 3 above)
Comment (alexey-pelykh) [reply hr_employee_id/models/hr_employee.py]: done

File: hr_employee_id/__manifest__.py (hunk 2 above)
Comment (alexey-pelykh) [reply hr_employee_id/__manifest__.py:20]: done

File: hr_employee_id/__init__.py (hunk 1 above)
Comment (alexey-pelykh) [reply hr_employee_id/__init__.py]: done

File: hr_employee_id/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# © 2011, 2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
```
Comment (simahawk) [inline hr_employee_id/__manifest__.py]: replace :copyright:  w/ "Copyright"

File: hr_employee_id/models/res_config.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        sequence = self.env.user.company_id.employee_id_sequence
+        if not sequence:
+            sequence = self.env.ref('hr_employee_id.seq_employeeid_ref')
+        return sequence and sequence.id or False
+
+    company_id = fields.Many2one(
+        'res.company',
+        string='Company',
+        required=True,
+        default=lambda self: self.env.user.company_id)
```
Comment (simahawk) [inline hr_employee_id/models/res_config.py]: move parenthesis to this empty line

File: hr_employee_id/models/res_config.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    employee_id_gen_method = fields.Selection(
+        related='company_id.employee_id_gen_method',
+        default=_default_id_gen_method
```
Comment (simahawk) [inline hr_employee_id/models/res_config.py]: use lambda in both fields to not reference the object: `default=lambda self: self._default_xx()`

**Review by simahawk — CHANGES_REQUESTED** [review]
code review only, LG. Just some nitpicking adjustments...

File: hr_employee_id/models/res_config.py (hunk 8 above)
Comment (alexey-pelykh) [reply hr_employee_id/models/res_config.py]: done

File: hr_employee_id/models/res_config.py (hunk 7 above)
Comment (alexey-pelykh) [reply hr_employee_id/models/res_config.py]: done

File: hr_employee_id/__manifest__.py (hunk 6 above)
Comment (alexey-pelykh) [reply hr_employee_id/__manifest__.py]: done

**Comment by alexey-pelykh** [comment]
@nikul-serpentcs @simahawk @pedrobaeza question - the menu is only visible to `base.group_no_one`, and I really see no reason for that. Maybe drop that limitation? Or otherwise move that menu away from "HR Manager" role.

**Comment by pedrobaeza** [comment]
Which menu do you mean?

File: hr_employee_id/views/res_config_views.xml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+    <record id="action_human_resources_employeeid_config" model="ir.actions.act_window">
+        <field name="name">Configure Employee ID</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">hr.employeeid.config.settings</field>
+        <field name="view_mode">form</field>
+        <field name="target">inline</field>
+    </record>
+
+    <menuitem id="menu_account_config" name="Employee ID" parent="hr.menu_human_resources_configuration"
```
Comment (alexey-pelykh) [inline hr_employee_id/views/res_config_views.xml]: @pedrobaeza this menu :)

**Comment by alexey-pelykh** [comment]
[^ this](https://github.com/OCA/hr/pull/488/files#r228296359)

File: hr_employee_id/views/res_config_views.xml (hunk 9 above)
Comment (pedrobaeza) [reply hr_employee_id/views/res_config_views.xml]: Well, I figure out that in the mind of original developers, the configuration is quite a technical action, so they hide if you don't have technical permissions (which is in vanilla Odoo activated when on debug mode), but if you hate as most of us the technical mode (and enable it by default to all), you can remove it.

Comment (alexey-pelykh) [reply hr_employee_id/views/res_config_views.xml]: Well, from a usability point of view, we have system administrators and specific Head of Staff, and calling sysadmins to enable debug mode for configuration is quite-quite odd. I think I'll remove it

Comment (pedrobaeza) [reply hr_employee_id/views/res_config_views.xml]: OK, it's something nobody will miss for sure.

**Comment by alexey-pelykh** [comment]
Okies, done here, it seems.

**Comment by alexey-pelykh** [comment]
I've changed where those settings live and simplified code a bit

**Comment by alexey-pelykh** [comment]
@nikul-serpentcs @simahawk any other amendments to the code?

**Comment by alexey-pelykh** [comment]
need assistance: when a user w/o `hr.group_hr_user` opens employee card, an error occurs:
```
Uncaught TypeError: Cannot read property 'string' of undefined
/web/content/1440-c24e898/web.assets_backend.js:1616
Traceback:
TypeError: Cannot read property 'string' of undefined
    at Class._renderTagLabel (/web/content/1440-c24e898/web.assets_backend.js:1616:660)
    at Class._renderNode [as _super] (/web/content/1440-c24e898/web.assets_backend.js:1607:246)
    at Class._renderNode (/web/content/1440-c24e898/web.assets_backend.js:2782:50)
    at Class._renderNode (/web/content/1082-262ffb6/web.assets_common.js:3540:371)
    at Function._.map._.collect (/web/content/1082-262ffb6/web.assets_common.js:13:270)
    at Class._renderGenericTag (/web/content/1440-c24e898/web.assets_backend.js:1597:303)
    at Class._renderNode [as _super] (/web/content/1440-c24e898/web.assets_backend.js:1610:13)
    at Class._renderNode (/web/content/1440-c24e898/web.assets_backend.js:2782:50)
    at Class._renderNode (/web/content/1082-262ffb6/web.assets_common.js:3540:371)
    at Function._.map._.collect (/web/content/1082-262ffb6/web.assets_common.js:13:270)
```
caused by this line
```
<label for="identification_id" groups="hr.group_hr_user" class="oe_edit_only"/>
``` 
And I'm out of ideas what to do with it.

**Comment by pedrobaeza** [comment]
Instead of putting groups attribute to elements, add the group to the entire view.

**Comment by alexey-pelykh** [comment]
@pedrobaeza a sample or link to it would be much appreciated - I've tried everything I've managed to think of placing groups, t-if's everywhere.

**Comment by alexey-pelykh** [comment]
@pedrobaeza fixed! awesome, I should read more docs 💯 thanks for support!

**Comment by pedrobaeza** [comment]
Glad to know it

**Comment by alexey-pelykh** [comment]
@nikul-serpentcs @simahawk please review the requested changes

**Review by simahawk — APPROVED** [review]

**Comment by alexey-pelykh** [comment]
@nikul-serpentcs any extra feedback?

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM

**Pull request merged**